site/app/controllers/pdf/PDFController.php

```diff
@@ -195,28 +195,31 @@ public function showGraderPDFEmbedded(string $gradeable_id) {
             }
         }
         $active_version = $graded_gradeable->getAutoGradedGradeable()->getActiveVersion();
-        $annotation_path = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable_id, $id, $active_version);
+        $annotation_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), 'annotations', $gradeable_id, $id, $active_version);
         $annotation_jsons = [];
-        //Dir iterator needs the first file.
-        if (is_dir($annotation_path) && count(scandir($annotation_path)) > 2) {
-            $first_file = scandir($annotation_path)[2];
-            $annotation_path = FileUtils::joinPaths($annotation_path, $first_file);
-            if (is_file($annotation_path)) {
-                $dir_iter = new \DirectoryIterator(dirname($annotation_path . '/'));
-                foreach ($dir_iter as $fileinfo) {
-                    if (!$fileinfo->isDot()) {
-                        $no_extension = preg_replace('/\\.[^.\\s]{3,4}$/', '', $fileinfo->getFilename());
-                        $pdf_info = explode('_', $no_extension);
-                        $pdf_id = implode('_', array_slice($pdf_info, 0, -1));
-                        $grader_id = $pdf_info[count($pdf_info) - 1];
-                        if ($pdf_id . '.pdf' === $filename) {
-                            $annotation_jsons[$grader_id] = file_get_contents($fileinfo->getPathname());
-                        }
+        $file_path = $_POST['file_path'];
+        if (is_dir($annotation_dir)) {
+            foreach (scandir($annotation_dir) as $annotation_file) {
+                if (explode('_', $annotation_file)[0] === md5($file_path)) {
+                    $file_contents = file_get_contents(FileUtils::joinPaths($annotation_dir, $annotation_file));
+                    $annotation_decoded = json_decode($file_contents, true);
+                    if ($annotation_decoded !== null) {
+                        $grader_id = $annotation_decoded["grader_id"];
+                        $annotation_jsons[$grader_id] = $file_contents;
                     }
                 }
             }
         }
-        $this->core->getOutput()->renderOutput(['PDF'], 'showPDFEmbedded', $gradeable_id, $id, $filename, $_POST['file_path'], $annotation_jsons, false, $page_num);
+        $params = [
+            "gradeable_id" => $gradeable_id,
+            "id" => $id,
+            "file_name" => $filename,
+            "file_path" => $file_path,
+            "annotation_jsons" => $annotation_jsons,
+            "is_student" => false,
+            "page_num" => $page_num
+        ];
+        $this->core->getOutput()->renderOutput(['PDF'], 'showPDFEmbedded', $params);
```

Comment:
Bad merge. The signature for `showPDFEmbedded` was changed to accept individual parameters, not an amorphous array.

Reply:
Reverted the call to showPDFEmbedded  above to use individual parameters